app/data_models/answer_store.py

```diff
@@ -134,3 +136,30 @@ def remove_all_answers_for_list_item_id(self, list_item_id: str):
 
     def serialize(self):
         return list(self.answer_map.values())
+
+    def get_escaped_answer(self, answer_id, list_item_id):
```

Comment:
Maybe a bit neater, I am not a huge fan of over using the same var. I also managed to get a walrus in before James suggested it :)

```
          if answer := self.get_answer(answer_id, list_item_id):
            if isinstance(answer.value, list):
                escaped_list = []
                for list_item in answer.value:
                    escaped_list.append(escape(list_item)) if isinstance(
                        list_item, str
                    ) else escaped_list.append(list_item)
                return escaped_list

            if isinstance(answer.value, dict):
                escaped_dict = {}
                for key, value in answer.value.items():
                    escaped_dict[key] = (
                        escape(value) if isinstance(value, str) else value
                    )
                return escaped_dict

            return (
                escape(answer.value) if isinstance(answer.value, str) else answer.value
            )

        return None

```

Reply:
Wow, this is compacted!

Reply:
Now if only we weren't type checking in if statements 😢